script/downsize_uploads.rb

```diff
@@ -77,23 +32,72 @@
     new_file = false
   end
 
+  before = upload.filesize
   upload.filesize = File.size(path)
+  delta = upload.filesize - before
+  return if delta > 0
+
   upload.sha1 = sha1
   upload.width = w
   upload.height = h
   upload.thumbnail_width = ww
   upload.thumbnail_height = hh
-  next unless upload.save!
 
   if new_file
-    next unless url = Discourse.store.store_upload(File.new(path), upload)
-    next unless upload.update!(url: url)
+    return unless url = Discourse.store.store_upload(File.new(path), upload)
+    upload.url = url
   end
 
-  upload.posts.each do |post|
-    post.update!(raw: post.raw.gsub(previous_short_url, upload.short_url)) if new_file
+  upload.save!
+
+  original_upload.posts.each do |post|
+    post.update!(raw: post.raw.gsub(original_upload.short_url, upload.short_url))
     Jobs.enqueue(:process_post, post_id: post.id, bypass_bump: true, cook: true)
   end
+
+  if new_file
+    Discourse.store.remove_upload(original_upload)
+  else
+    User.where(uploaded_avatar_id: original_upload.id).update_all(uploaded_avatar_id: upload.id)
```

Comment:
Are these the only ones where we use an `upload_id`? What about some of the uploads settings? What about the user profiles?

Reply:
Those were taken from [Upload's after_destroy](https://github.com/discourse/discourse/blob/master/app/models/upload.rb#L33-L37)
That probably should be updated as well.

Reply:
Yeah, have a look at [this query](https://github.com/discourse/discourse/blob/c75ebfee579e502f42cb7ea686ced91f31e654db/app/jobs/scheduled/clean_up_uploads.rb#L54-L80) I think it's the most up to date.